test/wasi/hello_world.wast

```diff
@@ -1,24 +1,23 @@
 (module
-  (import "wasi_snapshot_preview1" "fd_write" (func $__wasi_fd_write (param i32 i32 i32 i32) (result i32))) 
+  (import "wasi_snapshot_preview1" "fd_write" (func $wasi_fd_write (param i32 i32 i32 i32) (result i32))) 
   (memory 1)
   
   (export "memory" (memory 0))
-  (export "_start" (func $_start))
+  (export "hello_world" (func $hello_world))
   (data (i32.const 0) "Hello World!\n")
 
-  (func $_start (result i32)
-    (i32.store (i32.const 24) (i32.const 13)) ;; Lenght of "Hello World!\n"
-    (i32.store (i32.const 20) (i32.const 0))  ;; memory offset of "Hello World!\n"
-    (call $__wasi_fd_write
+  (func $hello_world
+    (i32.store (i32.const 100) (i32.const 0))
+    (i32.store (i32.const 112) (i32.const 13))
+    (i32.store (i32.const 200) (i32.const 0))
+    (call $wasi_fd_write
           (i32.const 1)  ;;file descriptor
-          (i32.const 20) ;;offset of str offset
-          (i32.const 1)  ;;iovec length
-          (i32.const 30) ;;result offset
+          (i32.const 100) ;;offset of str offset
+          (i32.const 2)  ;;iovec length
+          (i32.const 200) ;;result offset
```

Comment:
result offset? Isn't this an input?

Reply:
The result(s) of wasi functions is stored in memory at a given location, as the "real" result is the exit code of the function and they do not use multiple results. This is how emcc creates a wasm binary too (with 4 arguments instead of 3 as the documentation would lead you to believe).